src/SourceBuild/content/test/Microsoft.DotNet.SourceBuild.SmokeTests/ArtifactsSizeTest.cs

```diff
@@ -0,0 +1,100 @@
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.DotNet.SourceBuild.SmokeTests;
+
+[Trait("Category", "SdkContent")]
+public class ArtifactsSizeTest : SmokeTests
+{
+    private static readonly string BaselineFilePath = BaselineHelper.GetBaselineFilePath($"ArtifactsSizes/{Config.TargetRid}.txt");
+    private static readonly Dictionary<string, long> BaselineFileContent = new Dictionary<string, long>();
+    private static readonly Regex BuildVersionPattern = new(@"\b\d+\.\d+\.\d+[-@](alpha|preview|rc|rtm)\.\d(\.\d+\.\d+)?\b");
+    private const int SizeThreshold = 25;
+
+
+    public ArtifactsSizeTest(ITestOutputHelper outputHelper) : base(outputHelper)
+    {
+        if(File.Exists(BaselineFilePath))
+        {
+            string[] baselineFileContent = File.ReadAllLines(BaselineFilePath);
+            foreach(string entry in baselineFileContent)
+            {
+                string[] splitEntry = entry.Split(':');
+                BaselineFileContent[splitEntry[0].Trim()] = long.Parse(splitEntry[1].Trim());
+            }
+        }
+        else
+        {
+            Assert.True(Directory.Exists(BaselineHelper.GetBaselineFilePath("ArtifactsSizes/")));
+            Assert.False(true, $"Baseline file `{BaselineFilePath}' does not exist. Please create the baseline file then rerun the test.");
+        }
+    }
+
+    [SkippableFact(new[] { Config.SourceBuiltArtifactsPathEnv, Config.SdkTarballPathEnv, Config.TargetRidEnv }, skipOnNullOrWhiteSpace: true)]
+    public void CompareArtifactsToBaseline()
+    {
+        Assert.NotNull(Config.SourceBuiltArtifactsPath);
+        Assert.NotNull(Config.SdkTarballPath);
+        Assert.NotNull(Config.TargetRid);
+
+        IEnumerable<TarEntry> artifactsTarEntries = Utilities.GetTarballContent(Config.SourceBuiltArtifactsPath).Where(entry => entry.EntryType == TarEntryType.RegularFile);
+        IEnumerable<TarEntry> sdkTarEntries = Utilities.GetTarballContent(Config.SdkTarballPath).Where(entry => entry.EntryType == TarEntryType.RegularFile);
+
+        (string FilePath, long Bytes)[] tarEntries = sdkTarEntries.Concat(artifactsTarEntries)
+            .Select(entry =>
+            {
+                string result = BaselineHelper.RemoveVersions(entry.Name);
+                result = BaselineHelper.RemoveRids(result);
+                result = BaselineHelper.RemoveNetTfmPaths(result);
+
+                return (FilePath: result, Bytes: entry.Length);
+
+            })
+            .OrderBy(entry => entry.FilePath)
+            .ToArray();
+
+        foreach (var entry in tarEntries)
+        {
+            if (!BaselineFileContent.ContainsKey(entry.FilePath))
+            {
+                OutputHelper.LogWarningMessage($"{entry.FilePath} does not exist in baseline. Adding it to the baseline file");
+            }
+            else
+            {
+                CompareFileSizes(entry.FilePath, entry.Bytes, BaselineFileContent[entry.FilePath]);
+            }
```

Comment:
```suggestion
            if (!BaselineFileContent.TryGetValue(entry.FilePath, out long baselineBytes))
            {
                OutputHelper.LogWarningMessage($"{entry.FilePath} does not exist in baseline. Adding it to the baseline file");
            }
            else
            {
                CompareFileSizes(entry.FilePath, entry.Bytes, baselineBytes);
            }
```

Reply:
Addressed in this [commit](https://github.com/dotnet/installer/pull/17233/commits/ed23432b0722b0e225d5eeed5dc3029ba035003a).